tests/playwright/src/model/pages/main-page.ts

```diff
@@ -94,4 +99,33 @@ export abstract class MainPage extends BasePage {
     }
     return undefined;
   }
+
+  async getRowsFromTableByStatus(status: string): Promise<Array<Locator>> {
+    await waitUntil(async () => await this.rowsAreVisible(), { sendError: false });
+
+    const table = this.content.getByRole('table');
+    if (table) {
```

Comment:
If condition does not have a purpose here, `table` will always be defined since it's a locator, I think you might be confusing the Locator object with the underlining web element, which might or might not exist.